.github/workflows/ci.yml

```diff
@@ -18,12 +18,7 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
-      - name: Setup JDK
-        uses: actions/setup-java@v4
-        with:
-          distribution: corretto
-          java-version: 11
```

Comment:
Since [.tool-versions](https://github.com/guardian/thrift-serializer/blob/main/.tool-versions) specifies Java 21, this is a change in the Java version used for compilation from 11 to 21, which I reckon requires a bit more testing. I think we should keep the version at 11 in this PR by leaving setup-java as-is and adding setup-sbt, instead of using setup-scala. (Once we've updated this to Java 21 we can then move to setup-scala.)

Reply:
As @kelvin-chappell pointed out on another PR, the fact that this repo [is using the gha-scala-release-workflow](https://github.com/guardian/thrift-serializer/blob/main/.github/workflows/release.yml) means that the java version in .tool-versions is already being used to build the library!

So this only changes what's used for tests in CI: since those have passed, we’re fine to merge this PR as-is.

(We should look into whether we need to add a `--release` flag or similar to support older java versions among our consumers.)